lib/http-bridge/test/integration/Test/Integration/Scenario/Transactions.hs

```diff
@@ -309,3 +353,372 @@ spec = do
+                    "payments": [{
+                        "address": #{addr},
+                        "amount": {
+                            "quantity": 1,
+                            "unit": "lovelace"
+                        }
+                    }],
+                    "passphrase": "cardano-wallet"
+                }|]
+            r <- request @(ApiTransaction HttpBridge) ctx (postTx wSrc) Default payload
+            verify r
+                [ expectResponseCode HTTP.status400
+                , expectErrorMessage errMsg
+                ]
+
+    it "TRANS_CREATE_05 - [] as address" $ \ctx -> do
+        wSrc <- fixtureWallet ctx
+        let payload = Json [json|{
+                "payments": [{
+                    "address": [],
+                    "amount": {
+                        "quantity": 1,
+                        "unit": "lovelace"
+                    }
+                }],
+                "passphrase": "cardano-wallet"
+            }|]
+        r <- request @(ApiTransaction HttpBridge) ctx (postTx wSrc) Default payload
+        verify r
+            [ expectResponseCode HTTP.status400
+            , expectErrorMessage "expected Text, encountered Array"
+            ]
+
+    it "TRANS_CREATE_05 - Num as address" $ \ctx -> do
+        wSrc <- fixtureWallet ctx
+        let payload = Json [json|{
+                "payments": [{
+                    "address": 123123,
+                    "amount": {
+                        "quantity": 1,
+                        "unit": "lovelace"
+                    }
+                }],
+                "passphrase": "cardano-wallet"
+            }|]
+        r <- request @(ApiTransaction HttpBridge) ctx (postTx wSrc) Default payload
+        verify r
+            [ expectResponseCode HTTP.status400
+            , expectErrorMessage "expected Text, encountered Num"
+            ]
+
+    it "TRANS_CREATE_05 - address param missing" $ \ctx -> do
+        wSrc <- fixtureWallet ctx
+        let payload = Json [json|{
+                "payments": [{
+                    "amount": {
+                        "quantity": 1,
+                        "unit": "lovelace"
+                    }
+                }],
+                "passphrase": "cardano-wallet"
+            }|]
+        r <- request @(ApiTransaction HttpBridge) ctx (postTx wSrc) Default payload
+        verify r
+            [ expectResponseCode HTTP.status400
+            , expectErrorMessage "key 'address' not present"
+            ]
+
+    describe "TRANS_CREATE_06 - Invalid amount" $ do
+        let unitErr = "failed to parse quantified value. Expected value in\
+            \ 'lovelace' (e.g. { 'unit': 'lovelace', 'quantity': ... }"
+        let matrix = [ ( "Quantity = 0"
+                       , [json|{"quantity": 0, "unit": "lovelace"}|]
+                       , [ expectResponseCode HTTP.status500
+                         , expectErrorMessage "That's embarassing. It looks\
```

Comment:
Is this expected? Perhaps for `0` amount transaction the error code and message could be less _severe_ (e.g. HTTP 400 and message saying that 0 is not allowed...)

Reply:
500 is never expected. This is a mistake from our side.

Reply:
Minor thing also: careful about indentation on the matrix here :grimacing:

Reply:
Looking more carefully into the failure here, this is a very gray area at the moment. It shouldn't be forbidden to make transaction with no amount... At least "conceptually"; but current cardano-sl nodes don't really like that. In Shelley, it should be okay, so it's not that straightforward to just "fix" our transaction builder. 

Still, since the transaction builder is currently defined by the backend target, we can add a guard only for the bridge implementation, and leave it open in others.